pkgs/development/compilers/gcc/patches/default.nix

```diff
@@ -223,3 +223,7 @@ in
   }
   .${majorVersion} or [ ]
 )
+
+++ optional targetPlatform.isWindows [
```

Comment:
Do we need this for MinGW too? Seems like working around more a configuration screw up in GCC than anything intrinsic to the Windows and 32-bit.

Reply:
> Do we need this for MinGW too?

In practice, probably not, but I think that's only because nothing in mingw needs pre-libc gcc with C++ support.  Cygwin uses C++, which is probably unusual for pre-libc stuff.

Reply:
It would be good to have this as a comment in the code, then. and maybe name the patch accordingly too. (e.g. `pre-libc-cxx-support.patch` or something)